src/com/maddyhome/idea/vim/KeyHandler.java

```diff
@@ -200,67 +208,54 @@ public void handleKey(@NotNull Editor editor,
     // All the editor actions should be performed with top level editor!!!
     // Be careful: all the EditorActionHandler implementation should correctly process InjectedEditors
     editor = HelperKt.getTopLevelEditor(editor);
+
     final CommandState editorState = CommandState.getInstance(editor);
+    final CommandBuilder commandBuilder = editorState.getCommandBuilder();
 
     // If this is a "regular" character keystroke, get the character
     char chKey = key.getKeyChar() == KeyEvent.CHAR_UNDEFINED ? 0 : key.getKeyChar();
 
     final boolean isRecording = editorState.isRecording();
     boolean shouldRecord = true;
 
-    // Check for command count before key mappings - otherwise e.g. ':map 0 ^' breaks command counts that contain a zero
-    if (isCommandCount(editorState, chKey)) {
-      // Update the count
-      count = count * 10 + (chKey - '0');
-    }
-    else if (!waitCommandFinish(editor) && allowKeyMappings && handleKeyMapping(editor, key, context)) {
-      if (editorState.getMappingMode() != MappingMode.OP_PENDING ||
-          currentCmd.isEmpty() ||
-          currentCmd.peek().getArgument() == null ||
-          Objects.requireNonNull(currentCmd.peek().getArgument()).getType() != Argument.Type.OFFSETS) {
+    if (allowKeyMappings && handleKeyMapping(editor, key, context)) {
+      if (!editorState.isOperatorPending() || commandBuilder.peekCurrentCommandPartArgumentType() != Argument.Type.OFFSETS) {
```

Comment:
I'd like to get rid off `peekCurrentCommandPartArgumentType` if possible, and replace it with something a little more semantic, but I don't understand what's going on here - why are we returning at this point?

Reply:
This is a part of the logic that allows plugins to provide custom text objects. It's a bit messy right now, but since it should be updated to implement `:normal` command, I'll take care of this method.